packages/skygear-core/src/client.test.ts

```diff
@@ -0,0 +1,71 @@
+/* global jest */
+import { BaseAPIClient } from "./client";
+
+describe("APIClientDelegate", () => {
+  it("call onAccessTokenExpired on app launches", async (): Promise<void> => {
```

Comment:
> It is good semantics to say access token is 'expired' when app just launched? At that moment, we don't have any access token (they don't persist across app launches), so we're saying an non-existing access token is expired.

We do store the access token. Agree that it may not be good to say the access token is expired on app launch. But adding one more callback seems not good neither. Maybe rename `onAccessTokenExpired` to `refreshAccessToken`?

> It seems we're not using default delegate to refresh access tokens at the moment.

Why would you say so?

Reply:
> We do store the access token.

I thought access token should be stored in-memory only?

> Agree that it may not be good to say the access token is expired on app launch. But adding one more callback seems not good neither. Maybe rename onAccessTokenExpired to refreshAccessToken?

I think `refreshAccessToken` is good.

> It seems we're not using default delegate to refresh access tokens at the moment.

I mean are we not updating to do this in this PR?

Reply:
Added 2 commits to address this.